### ISSUE TITLE
Chrome: Playback stops at end of song

### ISSUE THREAD

**Issue by TheMagican** [airsonic/airsonic#705]
## Problem description
When playing songs in Chrome using the webinterface, the playback stops/hangs at the end of the song instead of playing the next song.
 * There are no errors in the server logs
 * There are no errors in Chrome Dev Console
 * I think this is a problem with _some_ constant-bitrate files (in my case MP3): I imported several albums with one media library scan. The VBR MP3 get played fine, the constant bitrate MP3s stop/hang at end of the track. I can not find a definitive attribute for an MP3 which produces the error, as other CBR MP3s that I imported some time ago do not produce the issue. (I can provide some MP3s that produce the error)
 * In the Playlist, the length of the song is given correctly (the same length that VLC says). When playing the song, the length given in the media control bar is 1sec longer
 * With "Stops/Hangs" I mean that the player stops at e.g. 3:31 (=the real duration of the song) while the length in the media control bar is reported as 3:32.
 * I had a look into the Airsonic DB: The song length is correct
 * The problem does not happen in Firefox: The length of the songs are correct, even in the media control bar, and at the end of a song, the next one is played.
 * The problem is reproducible with
   * Airsonic 10.1.1-RELEASE – 16. Dezember 2017
   * Subsonic 6.1.3 (build e408c9) – 31. Oktober 2017
   * Jamstash HTML Player
 * I am unsure, when the problem started, but I _think_ it started just some days/weeks ago, potentially with a Chrome update?

### Steps to reproduce

1. Create a playlist with some VBR and some CBR MP3s (in my case 320kbps)
2. Start playing a VBR file from start to end (or skip to near the end) --> At the end of the song, the next song is played automatically
3. Start playing a CBR file
   1. The song is reported 1sec longer in the media control bar
   2. The playback stops at the real length of the song, no matter if you play from start to end or skip to near the end of the song. 
   3. You have to skip manually to the next song.

## System information

 * **Airsonic version**: 10.1.1-RELEASE – 16. Dezember 2017
 * **Operating system**: Kubuntu 17.10
 * **Proxy server**: Aache 2.4.27-2ubuntu3
 * **Client**: Chrome 64.0.3282.186, 64bit

### Workaround
In the Airsonic settings:
1. Enable transcoding for the web player
2. In the transcoding settings, add "mp3" to the "convert from" list to enable a re-transcoding from MP3 to MP3

### Links
This issue is potentially related to #685

**Comment by dsm1212** [comment]
I've got a very similar problem with an album. It recurs on chrome or plex cast to chromecast. The songs stall one second prior to the end. BUT, the songs are 320kbps constant bitrate. The duration on the subsonic ui matches the duration in mediainfo for the song. For example, both are 3:21, the song stalls at 3:20. Every song on this album does it. This album worked at one time and I'm 100% certain it started well after my last subsonic update. So I think it was somehow brought out by a chrome update that must also affect chromecast.

**Comment by lord-carlos** [comment]
I have the same issue. Using latest docker file and playing flac files.

**Comment by plasmite** [comment]
Echoing this issue--the Chrome dev console shows the following when a song transition is expected:

Failed to load resource: net::ERR_CONTENT_LENGTH_MISMATCH       stream?player=2&id=4271:1
Failed to load resource: net::ERR_CONTENT_LENGTH_MISMATCH       stream:1

The workaround above does not work for FLAC files, as you might expect.

**Comment by tari** [comment]
I agree, looks like (another) duplicate of #685 where I've been exploring possible solutions.

**Comment by raaaste** [comment]
Not sure if this is helpful or not, but I'm using ffmpeg for transcoding (as per the instructions here https://airsonic.github.io/docs/transcode/), and disabling transcoding by removing the link to ffmpeg seems to fix the problem. This ofcourse disables transcoding altogether, so it's only a temporary solution.

The workaround mentioned in the original post did not help in my case. I tried playing both VBR and CBR MP3s, but the playback still cut off and I got the same errors @plasmite mentioned in their post.

My setup is
* Ubuntu 18.04
* tomcat 8
* airsonic 10.1.2-RELEASE
* apache acting as a reverse proxy, routing https traffic to the tomcat http port

**Comment by jangrewe** [comment]
I have the same issue, on Firefox 66.0.5 with Airsonic 10.3.1, using the web player.
Previously i added songs to a playlist and it happily played one after another. Now after every song it advances to the next song, but doesn't start playing it - so i need to go to the tab and click on the "play" icon in the web player after every song.

**Comment by jooola** [comment]
This also happen on Firefox, I had this bug. It seems it is not on every songs, for me at least.

**Comment by agodlydeciple** [comment]
Having the same issue here with Firefox 67.0 and Airsonic 10.3.1 being accessed via a reverse proxy.  I have also disabled transcoding altogether and nothing changed.

**Comment by tari** [comment]
This may be fixed with #1035, scheduled to be included in 10.4. If you're capable to building a snapshot and running that, it would be helpful to see if this is the same or a different issue.

**Comment by agodlydeciple** [comment]
@tari it appears to still be an issue with the 10.4-Snapshot version.  Let me know if there is any information you would like me to provide.

**Comment by fxthomas** [comment]
Hey @agodlydeciple, would you mind testing #1080? It seems to fix a similar issue on my end, but I'd like to have more tests on it.

**Comment by lemn8** [comment]
Updated to 10.3.1 i got this problem. It goes to the next song but stay's on 0:00. You can press play though. When i returned back to my previous .war (standalone version 10.2.1) everything is working fine again. For me i will just ignore the "new version 10.3.1" until the next. I really appreciate the really nice work you all have done so far it's a beautiful piece of software!

**Comment by agodlydeciple** [comment]
@fxthomas Getting to this now.  Let me check it out.

**Comment by agodlydeciple** [comment]
@fxthomas how would i pull your specific fix into the code base since it is waiting on a merge, so that I can test it for you?

**Comment by fxthomas** [comment]
Add my fork as a remote and switch to my branch, this should do the trick:

    git remote add fxthomas https://github.com/fxthomas/airsonic
    git fetch fxthomas
    git checkout try-to-fix-playback-intermittent-stops

**Comment by nx6** [comment]
> Updated to 10.3.1 i got this problem. It goes to the next song but stay's on 0:00. You can press play though. When i returned back to my previous .war (standalone version 10.2.1) everything is working fine again. 

Same story here -- except I'm using Waterfox as my browser instead of Chrome. Same fix: Stopped the FreeNAS jail I'm installed in, swapped back to 10.2.1 and tried again on the playlist I was listening to... songs playing one after another again on their own.

**Comment by jamespoore** [comment]
Is there any update/movement on this issue?
